neuralcompression/functional/_estimate_tails.py

```diff
@@ -0,0 +1,47 @@
+import typing
+
+import torch
+
+
+def estimate_tails(
+    cdf: typing.Callable[[torch.Tensor], torch.Tensor],
+    target: float,
+    shape: int,
+    device: typing.Optional[typing.Union[int, str, torch.device]] = None,
+    dtype: torch.dtype = torch.float32,
+) -> torch.Tensor:
```

Comment:
Primary ask is a docstring here describing what the function is doing mathematically - even if it's for internal use.